src/guide/agent-functions/interacting-with-environment.rst

```diff
@@ -48,10 +58,25 @@ Environmental macro properties can be read, via the returned :class:`DeviceMacro
         // Other behaviour code
         ...
     }
+
+  .. code-tab:: py Agent Python
+
+    @pyflamegpu.agent_function
+    def ExampleFn(message_in: pyflamegpu.MessageNone, message_out: pyflamegpu.MessageNone):
+        # Get the single float from environment macro property 'float1' and store it in local variable 'test_float'
+        test_float = FLAMEGPU->environment.getMacroPropertyFloat("float1")
+        # Get the root of the 3x3x3x3 environment macro property 'big_prop' and store it in a variable of the same name
+        # The dimensionality of the property is specified as arguments AFTER the variable name
+        bigprop = pyflamegpu.environment.getMacroPropertyInt("big_prop", 3, 3, 3, 3)
+        # Copy the value from location [1,1,1,1] to the variable t
+        t = big_prop[1][1][1][1]
+
+        # Other behaviour code
+        ...
     
 They can also be updated with a selection of functions, which execute atomically. These functions will update a single variable and return information related to it's old or new state. This can be useful, for simple actions such as conflict resolution and counting. However, if a basic read is subsequently required, a separate host or agent function in a following layer must be used (otherwise there would be a race condition). If running with ``SEATBELTS`` error checking enabled, an exception should be thrown where potential race conditions are detected.
 
-Macro properties support the normal :func:`+<flamegpu::DeviceMacroProperty::operator+>`, :func:`-<flamegpu::DeviceMacroProperty::operator->`, :func:`+=<flamegpu::DeviceMacroProperty::operator+=>`, :func:`-=<flamegpu::DeviceMacroProperty::operator-=>`, :func:`++<flamegpu::DeviceMacroProperty::operator++>`, :func:`--<flamegpu::DeviceMacroProperty::operator-->` operations. They also have access to a limited set of additional functions, explained in the table below.
+Macro properties support the normal :func:`+<flamegpu::DeviceMacroProperty::operator+>`, :func:`-<flamegpu::DeviceMacroProperty::operator->`, :func:`+=<flamegpu::DeviceMacroProperty::operator+=>`, :func:`-=<flamegpu::DeviceMacroProperty::operator-=>`, :func:`++<flamegpu::DeviceMacroProperty::operator++>` (only C++ supports pre and post increment), :func:`--<flamegpu::DeviceMacroProperty::operator-->` (only C++ supports pre and post increment) operations. They also have access to a limited set of additional functions, explained in the table below.
```

Comment:
```suggestion
Macro properties support the normal :func:`+<flamegpu::DeviceMacroProperty::operator+>`, :func:`-<flamegpu::DeviceMacroProperty::operator->`, :func:`+=<flamegpu::DeviceMacroProperty::operator+=>`, :func:`-=<flamegpu::DeviceMacroProperty::operator-=>`, :func:`++<flamegpu::DeviceMacroProperty::operator++>` (only C++ supports pre and post increment), :func:`--<flamegpu::DeviceMacroProperty::operator-->` (only C++ supports pre and post decrement) operations. They also have access to a limited set of additional functions, explained in the table below.
```

Reply:
Could make more terse additionaly via `C++ only` or similar, given they are next to the associated operators.